docs/markdown/Python/python-goals/python-check-goal.md

```diff
@@ -186,6 +186,30 @@ MY_SETTING = URLPattern(pattern="foo", callback=lambda: None)
 python_source(name="django_settings", source="django_settings.py")
 ```
 
+> 🚧 Importing from `extra_type_stubs`
+>
+> Requirements specified in `[mypy].extra_type_stubs` are not visible to the `python-infer` subsystem, and cannot be referenced as explicit `dependencies`. If you `import` from a stubs module in your code, and it does not have a corresponding implementation `python_requirement` target that provides the imported module, you may see a warning/error depending on the value you've configured for `[python-infer].unowned_dependency_behavior`. Goals other than `check` will also raise `ImportError`s if the `import` isn't conditional on the value of `typing.TYPE_CHECKING`:
+>
+> ```toml pants.toml
+> [python-infer]
+> unowned_dependency_behavior = "warning"
+>
+> [mypy]
+> extra_type_stubs = ["mypy-boto3-ec2==1.20.49"]
+> ```
+> ```python src/example.py
+> from typing import TYPE_CHECKING
+>
+> # Unsafe! Will fail outside of `check`
+> from mypy_boto3_ec2 import EC2Client
+>
+> if TYPE_CHECKING:
+>     # Safe, but will be flagged as a warning
+>     from mypy_boto3_ec2 import EC2ServiceResource
+> ```
+>
+> For these reasons, it's recommended to load any type-stub libraries that require explicit imports as part of your normal [third-party dependencies](doc:python-third-party-dependencies). Alternatively, you can set `# pants: no-infer-dep` on the lines of type-stub imports "guarded" by a check of `if TYPE_CHECKING`.
```

Comment:
It would be nice to explicitly mention the [type_stub_modules_mapping](https://www.pantsbuild.org/docs/reference-python_requirements#codetype_stubs_module_mappingcode) here, since I _think_ that's the right way to handle these